jenkinsci/jenkins
# [JENKINS-69113] renovate progressBar

The progressbar used at various places is replaced by a div/span combination and gets a new styling that is in sync with other ui elements. It is a little bit higher, has rounded corners and some padding, the minwidth is set to 10% so we always have at least a bit to show, it also avoids that the border is not rounded, e.g. when we're at just 2% and the progressbar is only 2px wide.
Animation is now based on css and no longer a gif and can also be enabled via a flag.
The build progress bar shown on run and console views is now updated dynamically including the tooltip.
The progress bar used in progressive rendering is doubled in size to make it more prominent that it is still running (See [JENKINS-72138], this doesn't solve the problem but might lower the chance that people reload the mentioned page because they think nothing happens).
This also fixes JENKINS-68631

**After** 
Build history widget:
<img width="242" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/0dcd8d1c-ca0f-4725-b596-4bba0bf59238">

Executor widget:
<img width="255" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/c7c363af-9760-4a18-b751-3353569add81">

Run view:
Here the progress bar will change over time
<img width="613" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/c78dc407-5ecd-4e7c-a560-1a87f5af5926">
<img width="309" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/7ba2547e-1c63-4425-b77d-196cecb3b459">


Console view with stuck:
<img width="608" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/ee84e5c4-b658-4fc3-b973-e0677cfda08c">

Progressive rendering (agent build history) with bigger progress bar:
<img width="597" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/2c067807-45e7-419f-bfaf-d3c7436b7016">






See [JENKINS-69113](https://issues.jenkins.io/browse/JENKINS-69113).



### Testing done
Manual testing:
- create job that runs for 1 minute
- trigger job (no previous builds) -> progress bar is at 100% and animated (on executor widgets, run page and console page)
- trigger job (with previous builds) -> progress bar is animated and increases over time

To see stuck jobs
- Create job that does nothing
- run the job several times
- Change job to sleep for 1 minute
- run job -> the progressbar should now be red animated.

Tested for freestyle and pipeline jobs



### Proposed changelog entries

- JENKINS-69113, renovate progressBar



### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
```

### Desired reviewers

@mention



Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
```
